### DISCUSSION TITLE
How to add filter actions before the resource is written to the storage layer?

### DISCUSSION THREAD

**Discussion by 1120151792** [clusterpedia-io/clusterpedia#543] [General]
I've been reading clusterpedia's documentation recently, but I didn't find out how to do a custom filter for a certain type of resources, so I want to expand some operations myself, so how can I extend clusterpedia to add some filtering actions before the resource is written to the storage layer?

**Comment by Iceber** [comment]
You want to filter data written to the storage?
Currently all received resource data is processed by calling the [ResourceStorage](https://github.com/clusterpedia-io/clusterpedia/blob/7a688956ca45838d0c1f11afc156a2e7ba02dffa/pkg/storage/storage.go#L30) interface, if you don't want to save some resources to the storage component, you may need to use [a custom storage layer](https://clusterpedia.io/docs/advanced-features/custom_storage_layer/) to filter them based on certain fields (label/annotation).
